### PR TITLE
Main including jenkins config

### PR DESCRIPTION
**Pull request by murakami10**
# ! Plz check only directory `jenkins`
other change is reviewed in this PR: 

## what
Add jenkins config: https://github.com/moneyforward/sakura_jenkins_config/pull/1

## issue
https://github.com/moneyforward/platform-sre-roadmap/issues/3082

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,13 @@
...
+## how to push image
+```shell
+export QUAY_USERNAME=your_username
+export QUAY_PASSWORD=your_password
+
+cd jenkins
+sudo docker login -u "$QUAY_USERNAME" -p "$QUAY_PASSWORD" quay.io
+sudo docker image build --build-arg ENV=dev -t quay.io/moneyforward/sakura_jenkins:test-1 .
+sudo docker push quay.io/moneyforward/sakura_jenkins:test-1
+```
```
Comment (murakami10) [inline README.md:13]: Ingore files except folder `jenkins`

File: jenkins/jcasc/dev.yml [hunk 2]
```diff
@@ -0,0 +1,141 @@
+credentials:
+  system:
+    domainCredentials:
+      - credentials:
+          - gitHubApp:
+              appID: "1069816"
+              description: "Gtihub App"
+              id: "github-app"
+              privateKey: "${readFile:/etc/jenkins/secrets/github-key.pem}"
```
Comment (murakami10) [inline jenkins/jcasc/dev.yml:9]: This credential is used to fetch Jenkinsfile from github repository.
Credential is set by ansible_ubuntu_provisioning

File: jenkins/jcasc/dev.yml [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+          - vaultTokenCredential:
+              description: "vault-token"
+              id: "vault-token"
+              scope: GLOBAL
+              token: "${json:vault-token:${readFile:/etc/jenkins/secrets/vault-token.json}}"
```
Comment (murakami10) [inline jenkins/jcasc/dev.yml:14]: This is used to fetch secrets from valut.
Credential is set by ansible_ubuntu_provisioning.

File: jenkins/jcasc/dev.yml [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+jenkins:
```
Comment (murakami10) [inline jenkins/jcasc/dev.yml:15]: Basically below config is the default config.
I added a comment to config that I modified intentionally

File: jenkins/jcasc/dev.yml [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+          properties:
+            - "apiToken"
+            - "consoleUrlProvider"
+            - preferredProvider:
+                providerId: "default"
+            - "myView"
+            - "timezone"
+            - "experimentalFlags"
+            - mailer:
+                emailAddress: "murakami.kyoya@moneyforward.co.jp"
```
Comment (murakami10) [inline jenkins/jcasc/dev.yml:65]: This is config for login.
In the future, I will create some users for each product.

Comment (murakami10) [reply jenkins/jcasc/dev.yml:65]: Password is hashed.

File: jenkins/jenkins.Dockerfile [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+USER jenkins
+
+COPY plugins.txt /usr/share/jenkins/ref/plugins.txt
+RUN jenkins-plugin-cli --plugin-file /usr/share/jenkins/ref/plugins.txt
+
+ARG ENV
+COPY jcasc/${ENV}.yml /var/jenkins_conf/jenkins.yaml
+ENV CASC_JENKINS_CONFIG=/var/jenkins_conf/jenkins.yaml
+ENV JAVA_OPTS="-Djenkins.install.runSetupWizard=false"
```
Comment (murakami10) [inline jenkins/jenkins.Dockerfile:15]: I will create CI/CD for pushing container image.

**Review by aqaurius6666 — APPROVED** [review]
lgtm

**Pull request merged**